### PR TITLE
[GPU] Resolve reduce test failure

### PR DESCRIPTION
**Pull request by byungilm**
+ Added exception logic to select planar format for blocked format
+ Bugfix of reorder_unreduced_axis_no_fusion
+ Bugfix to resolve an issue of cgan_mnist

Signed-off-by: Min, Byungil <byungil.min@intel.com>

### Details:
 - *item1*
 - *...*

### Tickets:
 - 67491

### CODE REVIEW

**Comment by byungilm** [comment]
@yeonbok Applied fix for cgan_mnist error that we discussed.

**Review by jade-cho — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/graph_optimizer/concat_input_order.cpp [hunk 1]
```diff
@@ -120,7 +120,8 @@ void concat_input_order::run(program& p) {
         // 4. Not already aligned
         // 5. Users can accept shuffled features
         // 6. No fused primitives
-        if (!node->is_type<concatenation>() || node->is_output() || (node->is_valid_output_layout() && node->is_dynamic()))
```
Comment (byungilm) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/concat_input_order.cpp:123]: fix related to dynamic shape

**Comment by isanghao** [comment]
Let me check performance

**Comment by isanghao** [comment]
No perf regression @ 2022-11-04--06-20

**Pull request merged**